### ISSUE TITLE
Threading problems in pre-release build

### ISSUE THREAD

**Issue by mlucy** [rethinkdb/rethinkdb#4910]
For @danielmewes:

```
>>> Failed cfeed_union.rb1.9 with exit code 1 after 8.0 sec + 0.0 sec setup (T+ 8.8 sec)

Server found crashed after test

Server ouput:
========== Start Create Console ===========
Recursively removing directory /home/ssd2/mlucy/rethinkdb/test/rql_test/.rethinkdbTestTem\
p-WqxHMq/.scratch-F5ctAb/cluster-oVAcPz/db-0/tmp
=========== End Create Console ============

Running rethinkdb 2.1.4-24-g30fe9d (debug) (CLANG 3.2 (tags/RELEASE_32/final))...
Running on Linux 3.2.0-89-generic x86_64
Loading data from directory /home/ssd2/mlucy/rethinkdb/test/rql_test/.rethinkdbTestTemp-W\
qxHMq/.scratch-F5ctAb/cluster-oVAcPz/db-0
Our server ID is a5460059-c358-4c8c-8d69-33aa084a11c1
Listening for intracluster connections on port 53746
Listening for client driver connections on port 32849
Listening for administrative HTTP connections on port 48545
Listening on addresses: 127.0.0.1, 127.0.1.1, 10.0.3.1, 192.168.122.1, 192.168.0.7, 192.1\
68.1.7, ::1, fe80::225:90ff:fe0d:8e8a%2, fe80::225:90ff:fe0d:8e8b%3, fe80::2cf1:ff:fefd:d\
9db%4
Server ready, "node_0" a5460059-c358-4c8c-8d69-33aa084a11c1
Version: rethinkdb 2.1.4-27-g7cb70b (debug) (CLANG 3.2 (tags/RELEASE_32/final))
error: Error in ../src/threading.cc at line 27:
error: Assertion failed: [home_thread() == get_thread_id()]
error: Backtrace:
error: Thu Oct  1 15:31:23 2015

       1: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:100
       2: backtrace_t::backtrace_t() at backtrace.cc:203
       3: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
       4: format_backtrace(bool) at backtrace.cc:198
       5: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       6: home_thread_mixin_t::assert_thread() const at threading.cc:27
       7: thread_allocation_t::thread_allocation_t(thread_allocator_t*) at threading.cc:6\
9
       8: real_table_persistence_interface_t::load_multistore(uuid_u const&, metadata_fil\
e_t::read_txn_t*, scoped_ptr_t<multistore_ptr_t>*, signal_t*, perfmon_collection_t*) at t\
able_interface.cc:283
       9: real_table_persistence_interface_t::create_multistore(uuid_u const&, scoped_ptr\
_t<multistore_ptr_t>*, signal_t*, perfmon_collection_t*) at table_interface.cc:309
       10: multi_table_manager_t::on_action(signal_t*, uuid_u const&, multi_table_manager\
_timestamp_t const&, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_c\
onfig_t> const&, boost::optional<raft_member_id_t> const&, boost::optional<raft_persisten\
t_state_t<table_raft_state_t> > const&, mailbox_addr_t<void ()> const&) at multi_table_ma\
nager.cc:374
       11: std::_Mem_fn<void (multi_table_manager_t::*)(signal_t*, uuid_u const&, multi_t\
able_manager_timestamp_t const&, multi_table_manager_bcard_t::status_t, boost::optional<t\
able_basic_config_t> const&, boost::optional<raft_member_id_t> const&, boost::optional<ra\
ft_persistent_state_t<table_raft_state_t> > const&, mailbox_addr_t<void ()> const&)>::ope\
rator()(multi_table_manager_t*, signal_t*, uuid_u const&, multi_table_manager_timestamp_t\
 const&, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t> con\
st&, boost::optional<raft_member_id_t> const&, boost::optional<raft_persistent_state_t<ta\
ble_raft_state_t> > const&, mailbox_addr_t<void ()> const&) const at functional:550
       12: _ZNSt5_BindIFSt7_Mem_fnIM21multi_table_manager_tFvP8signal_tRK6uuid_uRK31multi\
_table_manager_timestamp_tN27multi_table_manager_bcard_t8status_tERKN5boost8optionalI20ta\
ble_basic_config_tEERKNSD_I16raft_member_id_tEERKNSD_I23raft_persistent_state_tI18table_r\
aft_state_tEEERK14mailbox_addr_tIFvvEEEEPS1_St12_PlaceholderILi1EES11_ILi2EES11_ILi3EES11\
_ILi4EES11_ILi5EES11_ILi6EES11_ILi7EES11_ILi8EEEE6__callIvJOS3_OS4_OS7_OSB_OSF_OSJ_OSP_OS\
U_EJLi0ELi1ELi2ELi3ELi4ELi5ELi6ELi7ELi8EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EE\
E+0x1d9 at functional:1145
       13: _ZNSt5_BindIFSt7_Mem_fnIM21multi_table_manager_tFvP8signal_tRK6uuid_uRK31multi\
_table_manager_timestamp_tN27multi_table_manager_bcard_t8status_tERKN5boost8optionalI20ta\
ble_basic_config_tEERKNSD_I16raft_member_id_tEERKNSD_I23raft_persistent_state_tI18table_r\
aft_state_tEEERK14mailbox_addr_tIFvvEEEEPS1_St12_PlaceholderILi1EES11_ILi2EES11_ILi3EES11\
_ILi4EES11_ILi5EES11_ILi6EES11_ILi7EES11_ILi8EEEEclIJS3_S4_S7_SB_SF_SJ_SP_SU_EvEET0_DpOT_\
+0x120 at functional:1203
       14: std::_Function_handler<void (signal_t*, uuid_u, multi_table_manager_timestamp_\
t, multi_table_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::o\
ptional<raft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >,\
 mailbox_addr_t<void ()>), std::_Bind<std::_Mem_fn<void (multi_table_manager_t::*)(signal\
_t*, uuid_u const&, multi_table_manager_timestamp_t const&, multi_table_manager_bcard_t::\
status_t, boost::optional<table_basic_config_t> const&, boost::optional<raft_member_id_t>\
 const&, boost::optional<raft_persistent_state_t<table_raft_state_t> > const&, mailbox_ad\
dr_t<void ()> const&)> (multi_table_manager_t*, std::_Placeholder<1>, std::_Placeholder<2\
>, std::_Placeholder<3>, std::_Placeholder<4>, std::_Placeholder<5>, std::_Placeholder<6>\
, std::_Placeholder<7>, std::_Placeholder<8>)> >::_M_invoke(std::_Any_data const&, signal\
_t*, uuid_u, multi_table_manager_timestamp_t, multi_table_manager_bcard_t::status_t, boos\
t::optional<table_basic_config_t>, boost::optional<raft_member_id_t>, boost::optional<raf\
t_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()>) at functional:1779
       15: std::function<void (signal_t*, uuid_u, multi_table_manager_timestamp_t, multi_\
table_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<r\
aft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_\
addr_t<void ()>)>::operator()(signal_t*, uuid_u, multi_table_manager_timestamp_t, multi_t\
able_manager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<ra\
ft_member_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_a\
ddr_t<void ()>) const at functional:2160
       16: mailbox_t<void (uuid_u, multi_table_manager_timestamp_t, multi_table_manager_b\
card_t::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_member_id_t\
>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()>\
)>::read_impl_t::read(read_stream_t*, signal_t*) at typed.hpp:705
       17: mailbox_manager_t::mailbox_read_coroutine(connectivity_cluster_t::connection_t\
*, auto_drainer_t::lock_t, threadnum_t, unsigned long, std::vector<char, std::allocator<c\
har> >*, long, mailbox_manager_t::force_yield_t) at mailbox.cc:277
       18: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x202f338] at 0x202f\
338 ()
       19: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x202f21c] at 0x202f\
21c ()
       20: callable_action_wrapper_t::run() at runtime_utils.cc:43
       21: coro_t::run() at coroutines.cc:205
       22: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x202e54d] at 0x202e\
54d ()
       23: mailbox_manager_t::on_local_message(connectivity_cluster_t::connection_t*, aut\
o_drainer_t::lock_t, std::vector<char, std::allocator<char> >&&) at mailbox.cc:212
       24: connectivity_cluster_t::send_message(connectivity_cluster_t::connection_t*, au\
to_drainer_t::lock_t, unsigned char, cluster_send_message_write_callback_t*) at cluster.c\
c:1345
       25: send(mailbox_manager_t*, raw_mailbox_t::address_t, mailbox_write_callback_t*) \
at mailbox.cc:129
       26: void send<uuid_u, multi_table_manager_timestamp_t, multi_table_manager_bcard_t\
::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_member_id_t>, boo\
st::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<void ()> >(mai\
lbox_manager_t*, mailbox_t<void (uuid_u, multi_table_manager_timestamp_t, multi_table_man\
ager_bcard_t::status_t, boost::optional<table_basic_config_t>, boost::optional<raft_membe\
r_id_t>, boost::optional<raft_persistent_state_t<table_raft_state_t> >, mailbox_addr_t<vo\
id ()>)>::address_t, uuid_u const&, multi_table_manager_timestamp_t const&, multi_table_m\
anager_bcard_t::status_t const&, boost::optional<table_basic_config_t> const&, boost::opt\
ional<raft_member_id_t> const&, boost::optional<raft_persistent_state_t<table_raft_state_\
t> > const&, mailbox_addr_t<void ()> const&) at typed.hpp:744
       27: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x1aedf22] at 0x1aed\
f22 ()
       28: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x1aedcbc] at 0x1aed\
cbc ()
       29: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x1aedc1c] at 0x1aed\
c1c ()
       30: callable_action_wrapper_t::run() at runtime_utils.cc:43
       31: coro_t::run() at coroutines.cc:205
       32: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x1aed9ad] at 0x1aed\
9ad ()
       33: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x1aed979] at 0x1aed\
979 ()
       34: /home/ssd2/mlucy/rethinkdb/build/debug_clang/rethinkdb() [0x1aeb356] at 0x1aeb\
356 ()
       35: table_meta_client_t::create_or_emergency_repair(uuid_u const&, table_raft_stat\
e_t const&, multi_table_manager_timestamp_t::epoch_t const&, signal_t*) at table_meta_cli\
ent.cc:566
       36: table_meta_client_t::create(uuid_u, table_config_and_shards_t const&, signal_t\
*) at table_meta_client.cc:284
       37: real_reql_cluster_interface_t::table_create(name_string_t const&, counted_t<ql\
::db_t const>, table_generate_config_params_t const&, std::string const&, write_durabilit\
y_t, signal_t*, ql::datum_t*, admin_err_t*) at real_reql_cluster_interface.cc:263
       38: artificial_reql_cluster_interface_t::table_create(name_string_t const&, counte\
d_t<ql::db_t const>, table_generate_config_params_t const&, std::string const&, write_dur\
ability_t, signal_t*, ql::datum_t*, admin_err_t*) at artificial_reql_cluster_interface.cc\
:81
       39: ql::table_create_term_t::eval_impl(ql::scope_env_t*, ql::args_t*, ql::eval_fla\
gs_t) const at db_table.cc:200
       40: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:22\
5
       41: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:\
350
       42: ql::query_cache_t::ref_t::run(ql::env_t*, Response*) at query_cache.cc:309
       43: ql::query_cache_t::ref_t::fill_response(Response*) at query_cache.cc:248
       44: ql::run(ql::query_id_t&&, ql::protob_t<Query>, Response*, ql::query_cache_t*, \
signal_t*) at term.cc:269
       45: rdb_query_server_t::run_query(ql::query_id_t&&, ql::protob_t<Query> const&, Re\
sponse*, ql::query_cache_t*, signal_t*) at query_server.cc:49
       46: void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsig\
ned long, ql::query_cache_t*, signal_t*)::query_server_t::connection_loop<json_protocol_t\
>::{lambda(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > >, signal_\
t*)#1}::operator()(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > >,\
 signal_t*) const::{lambda()#1}::operator()() const at protob.cc:616
       47: void save_exception<void query_server_t::connection_loop<json_protocol_t>(linu\
x_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::query_server_t::connection_\
loop<json_protocol_t>::{lambda(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t\
loop<json_protocol_t>::{lambda(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t\
<Query> > >, signal_t*)#1}::operator()(std::_List_iterator<std::pair<ql::query_id_t, ql::\
protob_t<Query> > >, signal_t*) const::{lambda()#1}>(std::__exception_ptr::exception_ptr*\
, std::string*, cond_t*, void query_server_t::connection_loop<json_protocol_t>(linux_tcp_\
conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::query_server_t::connection_loop<j\
son_protocol_t>::{lambda(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query\
> > >, signal_t*)#1}::operator()(std::_List_iterator<std::pair<ql::query_id_t, ql::protob\
_t<Query> > >, signal_t*) const::{lambda()#1}&&) at protob.cc:556
       48: void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsig\
ned long, ql::query_cache_t*, signal_t*)::query_server_t::connection_loop<json_protocol_t\
>::{lambda(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > >, signal_\
t*)#1}::operator()(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > >,\
 signal_t*) const at protob.cc:613
       49: std::_Function_handler<void (std::_List_iterator<std::pair<ql::query_id_t, ql:\
:protob_t<Query> > >, signal_t*), void query_server_t::connection_loop<json_protocol_t>(l\
inux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda(std::_List_itera\
tor<std::pair<ql::query_id_t, ql::protob_t<Query> > >, signal_t*)#1}>::_M_invoke(std::_An\
y_data const&, std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > >, sig\
nal_t*) at functional:1779
       50: std::function<void (std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t\
<Query> > >, signal_t*)>::operator()(std::_List_iterator<std::pair<ql::query_id_t, ql::pr\
otob_t<Query> > >, signal_t*) const at functional:2160
       51: std_function_callback_t<std::_List_iterator<std::pair<ql::query_id_t, ql::prot\
ob_t<Query> > > >::coro_pool_callback(std::_List_iterator<std::pair<ql::query_id_t, ql::p\
rotob_t<Query> > >, signal_t*) at coro_pool.hpp:30
       52: coro_pool_t<std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> \
> > >::worker_run(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > >, \
auto_drainer_t::lock_t) at coro_pool.hpp:68
       53: std::_Mem_fn<void (coro_pool_t<std::_List_iterator<std::pair<ql::query_id_t, q\
l::protob_t<Query> > > >::*)(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Q\
uery> > >, auto_drainer_t::lock_t)>::operator()(coro_pool_t<std::_List_iterator<std::pair\
<ql::query_id_t, ql::protob_t<Query> > > >*, std::_List_iterator<std::pair<ql::query_id_t\
, ql::protob_t<Query> > >, auto_drainer_t::lock_t) const at functional:550
       54: _ZNSt5_BindIFSt7_Mem_fnIM11coro_pool_tISt14_List_iteratorISt4pairIN2ql10query_\
id_tENS4_8protob_tI5QueryEEEEEFvSA_N14auto_drainer_t6lock_tEEEPSB_SA_SD_EE6__callIvJEJLi0\
ELi1ELi2EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE+0xa0 at functional:1145
       55: _ZNSt5_BindIFSt7_Mem_fnIM11coro_pool_tISt14_List_iteratorISt4pairIN2ql10query_\
id_tENS4_8protob_tI5QueryEEEEEFvSA_N14auto_drainer_t6lock_tEEEPSB_SA_SD_EEclIJEvEET0_DpOT\
_+0x26 at functional:1203
       56: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (coro_pool_t<std::_Lis\
t_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > > >::*)(std::_List_iterator<st\
d::pair<ql::query_id_t, ql::protob_t<Query> > >, auto_drainer_t::lock_t)> (coro_pool_t<st\
d::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > > >*, std::_List_iterat\
or<std::pair<ql::query_id_t, ql::protob_t<Query> > >, auto_drainer_t::lock_t)> >::run_act\
ion() at callable_action.hpp:31
       57: callable_action_wrapper_t::run() at runtime_utils.cc:43
       58: coro_t::run() at coroutines.cc:205
       59: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (coro_pool_t<std::\
_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > > >::*)(std::_List_iterato\
r<std::pair<ql::query_id_t, ql::protob_t<Query> > >, auto_drainer_t::lock_t)> (coro_pool_\
t<std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > > >*, std::_List_it\
erator<std::pair<ql::query_id_t, ql::protob_t<Query> > >, auto_drainer_t::lock_t)> >(std:\
:_Bind<std::_Mem_fn<void (coro_pool_t<std::_List_iterator<std::pair<ql::query_id_t, ql::p\
rotob_t<Query> > > >::*)(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query\
> > >, auto_drainer_t::lock_t)> (coro_pool_t<std::_List_iterator<std::pair<ql::query_id_t\
, ql::protob_t<Query> > > >*, std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<\
Query> > >, auto_drainer_t::lock_t)>&&) at coroutines.hpp:58
       60: coro_pool_t<std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> \
> > >::on_source_availability_changed() at coro_pool.hpp:88
       61: availability_control_t::set_available(bool) at passive_producer.hpp:65
       62: limited_fifo_queue_t<std::_List_iterator<std::pair<ql::query_id_t, ql::protob_\
t<Query> > >, std::list<std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query>\
 > >, std::allocator<std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > \
> > > >::push(std::_List_iterator<std::pair<ql::query_id_t, ql::protob_t<Query> > > const\
&) at limited_fifo.hpp:35
       63: void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsig\
ned long, ql::query_cache_t*, signal_t*)::query_server_t::connection_loop<json_protocol_t\
>::{lambda()#1}::operator()() const at protob.cc:650
       64: void save_exception<void query_server_t::connection_loop<json_protocol_t>(linu\
x_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#1}>(std::__except\
ion_ptr::exception_ptr*, std::string*, cond_t*, void query_server_t::connection_loop<json\
_protocol_t>(linux_tcp_conn_t*, unsigned long, ql::query_cache_t*, signal_t*)::{lambda()#\
1}&&) at protob.cc:556
       65: void query_server_t::connection_loop<json_protocol_t>(linux_tcp_conn_t*, unsig\
ned long, ql::query_cache_t*, signal_t*) at protob.cc:645
       66: query_server_t::handle_conn(scoped_ptr_t<linux_tcp_conn_descriptor_t> const&, \
auto_drainer_t::lock_t) at protob.cc:490
       67: std::_Mem_fn<void (query_server_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t\
> const&, auto_drainer_t::lock_t)>::operator()(query_server_t*, scoped_ptr_t<linux_tcp_co\
nn_descriptor_t> const&, auto_drainer_t::lock_t) const at functional:550
       68: _ZNSt5_BindIFSt7_Mem_fnIM14query_server_tFvRK12scoped_ptr_tI27linux_tcp_conn_d\
escriptor_tEN14auto_drainer_t6lock_tEEEPS1_St12_PlaceholderILi1EES8_EE6__callIvJRS4_EJLi0\
ELi1ELi2EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE+0x9d at functional:1145
       68: _ZNSt5_BindIFSt7_Mem_fnIM14query_server_tFvRK12scoped_ptr_tI27linux_tcp_conn_d\
escriptor_tEN14auto_drainer_t6lock_tEEEPS1_St12_PlaceholderILi1EES8_EE6__callIvJRS4_EJLi0\
ELi1ELi2EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE+0x9d at functional:1145
       69: _ZNSt5_BindIFSt7_Mem_fnIM14query_server_tFvRK12scoped_ptr_tI27linux_tcp_conn_d\
escriptor_tEN14auto_drainer_t6lock_tEEEPS1_St12_PlaceholderILi1EES8_EEclIJRS4_EvEET0_DpOT\
_+0x3d at functional:1203
       70: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), std:\
:_Bind<std::_Mem_fn<void (query_server_t::*)(scoped_ptr_t<linux_tcp_conn_descriptor_t> co\
nst&, auto_drainer_t::lock_t)> (query_server_t*, std::_Placeholder<1>, auto_drainer_t::lo\
ck_t)> >::_M_invoke(std::_Any_data const&, scoped_ptr_t<linux_tcp_conn_descriptor_t>&) at\
 functional:1779
       71: std::function<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&)>::operator()(s\
coped_ptr_t<linux_tcp_conn_descriptor_t>&) const at functional:2160
       72: linux_nonthrowing_tcp_listener_t::handle(int) at network.cc:924
       73: std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)>::operator()(linu\
x_nonthrowing_tcp_listener_t*, int) const at functional:550
       74: _ZNSt5_BindIFSt7_Mem_fnIM32linux_nonthrowing_tcp_listener_tFviEEPS1_iEE6__call\
IvJEJLi0ELi1EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE+0x6f at functional:1145
       75: _ZNSt5_BindIFSt7_Mem_fnIM32linux_nonthrowing_tcp_listener_tFviEEPS1_iEEclIJEvE\
ET0_DpOT_+0x26 at functional:1203
       76: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp\
_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >::run_action() at callab\
le_action.hpp:31
       77: callable_action_wrapper_t::run() at runtime_utils.cc:43
       78: coro_t::run() at coroutines.cc:205
       79: void coro_t::spawn_now_dangerously<std::_Bind<std::_Mem_fn<void (linux_nonthro\
wing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener_t*, int)> >(std::_Bind<std:\
:_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(int)> (linux_nonthrowing_tcp_listener\
_t*, int)>&&) at coroutines.hpp:52
       80: linux_nonthrowing_tcp_listener_t::accept_loop(auto_drainer_t::lock_t) at netwo\
rk.cc:891
       81: std::_Mem_fn<void (linux_nonthrowing_tcp_listener_t::*)(auto_drainer_t::lock_t\
)>::operator()(linux_nonthrowing_tcp_listener_t*, auto_drainer_t::lock_t) const at functi\
onal:550
       82: _ZNSt5_BindIFSt7_Mem_fnIM32linux_nonthrowing_tcp_listener_tFvN14auto_drainer_t\
6lock_tEEEPS1_S3_EE6__callIvJEJLi0ELi1EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE\
+0x7c at functional:1145
       83: _ZNSt5_BindIFSt7_Mem_fnIM32linux_nonthrowing_tcp_listener_tFvN14auto_drainer_t\
6lock_tEEEPS1_S3_EEclIJEvEET0_DpOT_+0x26 at functional:1203
       84: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (linux_nonthrowing_tcp\
_listener_t::*)(auto_drainer_t::lock_t)> (linux_nonthrowing_tcp_listener_t*, auto_drainer\
_t::lock_t)> >::run_action() at callable_action.hpp:31
       85: callable_action_wrapper_t::run() at runtime_utils.cc:43
error: Exiting.
```

**Comment by danielmewes** [comment]
I can't find a way in which this could happen, unless maybe the `real_table_persistance_interface_t` gets destructed while a mailbox callback is still active (I can't see an obvious mechanism to stop that from happening). That's not super likely though, since it will only get destructed on shutting down the server.

I'll run the tests a few more times, but I don't think we need to block the release for this (even if this function is run on the wrong thread, it's actually very unlikely to lead to any crashes or incorrect behavior).

**Comment by mlucy** [comment]
I've also been unable to reproduce this, so it's possible my build got into a bad state.

**Comment by danielmewes** [comment]
Closing, we haven't seen this again afaik.
